contrib/minimize-emulation.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python3
+#
+# Copyright 2024 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# pylint: disable=invalid-name,missing-docstring
+
+import json
+import sys
+import hashlib
+import zipfile
+
+
+def _minimize_usb_events(data) -> None:
+    for device in data["UsbDevices"]:
+        try:
+            for event in device.get("UsbEvents", []) + device.get("Events", []):
+                event["Id"] = hashlib.sha1(event["Id"].encode()).hexdigest()[:8]
+        except KeyError:
+            pass
+
+
+if __name__ == "__main__":
+    for fn in sys.argv[1:]:
+        files = {}
+        with zipfile.ZipFile(fn) as myzip:
```

Comment:
if you follow my suggestion for pre-commit then I think you probably want to check the header of the file to determine whether to iterate.

Reply:
Good idea, done.